src/Hippo/Proxies/YarpReverseProxy.cs

```diff
@@ -12,14 +12,14 @@ public class YarpReverseProxy : IReverseProxy
     public YarpReverseProxy(ITaskQueue<ReverseProxyUpdateRequest> reverseProxyConfigQueue, IJobScheduler scheduler)
     {
         _reverseProxyConfigQueue = reverseProxyConfigQueue;
-        InternalScheduler s = scheduler as InternalScheduler;
+        JobScheduler s = scheduler as JobScheduler;
```

Comment:
If we have a unified JobScheduler base type, should we get rid of IJobScheduler and avoid the cast?  Or do we still need a distinct interface for other reasons?

Reply:
Still required for dependency injection IIRC. Is there a way to pass an abstract class into a DI constructor?

Reply:
Ah, makes sense.  Not sure about whether you can register an abstract class as a service (and struggling with the docs), but might be worth giving it a try.  No big deal though - thanks for clarifying!

Reply:
Ah! It turns out that you can use an abstract class for dependency injection! Fixed.